go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -0,0 +1,754 @@
+		namespaceLister:    namespaceInformer.Lister(),
+		namespaceSynced:    namespaceInformer.Informer().HasSynced,
+		podLister:          podInformer.Lister(),
+		podSynced:          podInformer.Informer().HasSynced,
+		routeManager:       routeManager,
+		linkManager:        linkmanager.NewController(nodeName, v4, v6),
+		ruleManager:        rulemanager.NewController(v4, v6),
+		iptablesManager:    iptables.NewController(v4, v6),
+		triggerReconcileCh: make(chan struct{}, 1),
+		nodeName:           nodeName,
+		v4:                 v4,
+		v6:                 v6,
+	}
+
+	for _, resourceInformer := range []struct {
+		resourceName string
+		cache.SharedIndexInformer
+	}{
+		{
+			"egressip",
+			eIPInformer.Informer(),
+		},
+		{
+			"node",
+			nodeInformer,
+		},
+		{
+			"namespace",
+			namespaceInformer.Informer(),
+		},
+		{
+			"pod",
+			podInformer.Informer(),
+		},
+	} {
+		_, err = resourceInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.onAdd,
+			UpdateFunc: c.onUpdate,
+			DeleteFunc: c.onDelete,
+		})
+		if err != nil {
+			return c, fmt.Errorf("failed to add event hander for resource %q: %v", resourceInformer.resourceName, err)
+		}
+	}
+	return c, nil
+}
+
+func (c *Controller) Run(stopCh <-chan struct{}, wg *sync.WaitGroup) error {
+	klog.Info("Starting Egress IP Controller")
+
+	for _, resource := range []struct {
+		name string
+		cache.InformerSynced
+	}{
+		{
+			"egressip",
+			c.egressIPSynced,
+		},
+		{
+			"node",
+			c.nodeSynced,
+		},
+		{
+			"namespace",
+			c.namespaceSynced,
+		},
+		{
+			"pod",
+			c.podSynced,
+		},
+	} {
+		if !cache.WaitForNamedCacheSync(resource.name, stopCh, resource.InformerSynced) {
+			return fmt.Errorf("timed out waiting for %q caches to sync", resource.name)
+		}
+	}
+	// tell rule manager that we want to fully own all rules at a particular priority. Any rules created with this priority
+	// ,and we do not recognize it, will be removed.
+	if err := c.ruleManager.OwnPriority(routeRulePriority); err != nil {
+		klog.Errorf("Egress IP controller: failed to own priority %d - stale entries maybe present: %v", routeRulePriority, err)
+	}
+	if c.v4 {
+		if err := c.iptablesManager.OwnChain(utiliptables.TableNAT, multiNICChain, utiliptables.ProtocolIPv4); err != nil {
+			return fmt.Errorf("unable to own chain %s: %v", multiNICChain, err)
+		}
+	}
+	if c.v6 {
+		if err := c.iptablesManager.OwnChain(utiliptables.TableNAT, multiNICChain, utiliptables.ProtocolIPv6); err != nil {
+			return fmt.Errorf("unable to own chain %s: %v", multiNICChain, err)
+		}
+	}
+
+	go func() {
+		var err error
+		for {
+			select {
+			case <-stopCh:
+				wg.Done()
+				return
+			case <-c.triggerReconcileCh:
+				if err = c.reconcile(); err != nil {
+					klog.Errorf("Failed to reconcile egress IP: %v", err)
+				}
+			}
+		}
+	}()
+
+	wg.Add(3)
+	go func() {
+		c.linkManager.Run(stopCh, 2*time.Minute)
+		wg.Done()
+	}()
+
+	go func() {
+		c.iptablesManager.Run(stopCh, 6*time.Minute)
```

Comment:
Total guess work :) I wanted to keep the intervals different for each component to spread the CPU spikes caused by their execution and then I thought what would be an ok interval to check if the user / app messed with what we are configuring while also not chewing up too much CPU because we are checking too often..